### PR TITLE
Add diagram for substations

### PR DESCRIPTION
**Pull request by FranckLecuyer**
This PR must be merged after : 
https://github.com/gridsuite/study-server/pull/52
https://github.com/powsybl/powsybl-single-line-diagram-server/pull/38

### CODE REVIEW

File: src/components/study-pane.js [hunk 1]
```diff
@@ -539,13 +576,40 @@ const StudyPane = (props) => {
...
-            useName && displayedVoltageLevel
+        let sldTitle;
+        let svgUrl;
+        if (displayedVoltageLevel) {
+            sldTitle = useName
                 ? displayedVoltageLevel.name +
                   ' \u002D ' +
                   network.getSubstation(displayedVoltageLevel.substationId)
                       .countryName
                 : displayedVoltageLevelId;
```
Comment (SlimaneAmar) [inline src/components/study-pane.js:587]: If the id is not present the tittle is 'undefined'

File: src/components/study-pane.js [hunk 2]
```diff
@@ -144,6 +147,8 @@ const StudyPane = (props) => {
         (state) => state.filteredNominalVoltages
     );
 
+    const [displayedSubstationId, setDisplayedSubstationId] = useState(null);
+
```
Comment (SlimaneAmar) [inline src/components/study-pane.js:151]: Are we obliged to manage 2 state variables : displayedSubstationId & displayedVoltageLevelId
They are exclusive( only one and only one SLD)

File: src/components/study-pane.js [hunk 3]
```diff
@@ -539,13 +576,40 @@ const StudyPane = (props) => {
...
+            svgUrl = getSubstationSingleLineDiagram(
+                studyName,
+                userId,
+                displayedSubstationId,
+                useName,
+                centerName,
+                diagonalName,
+                substationLayout
+            );
+        }
```
Comment (SlimaneAmar) [inline src/components/study-pane.js:612]: Can be factored (wtih substationLayout=null ) ?

**Review by SlimaneAmar — APPROVED** [review]

**Review by niconoir — APPROVED** [review]

File: src/components/single-line-diagram.js [hunk 4]
```diff
@@ -149,33 +165,62 @@ const SingleLineDiagram = forwardRef((props, ref) => {
...
             const svgEl = divElt.getElementsByTagName('svg')[0];
             const bbox = svgEl.getBBox();
             const xOrigin = bbox.x - 20;
             const yOrigin = bbox.y - 20;
             const svgWidth = bbox.width + 40;
             const svgHeight = bbox.height + 40;
+
+            let sizeWidth = svgWidth;
+            let sizeHeight = svgHeight;
+            if (svgType === 'substation') {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/single-line-diagram.js:198]: Done

File: src/components/single-line-diagram.js [hunk 5]
```diff
@@ -149,33 +165,62 @@ const SingleLineDiagram = forwardRef((props, ref) => {
...
+                                ? sizeWidth / 4
+                                : -Infinity,
+                        bottom:
+                            svgType === 'voltage-level'
+                                ? sizeHeight / 4
+                                : -Infinity,
+                        right:
+                            svgType === 'voltage-level'
+                                ? sizeWidth / 4
+                                : -Infinity,
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/single-line-diagram.js]: Done

File: src/components/study-pane.js (hunk 1 above)
Comment (FranckLecuyer) [reply src/components/study-pane.js]: Corrected

**Review by med-hedi — APPROVED** [review]

File: src/utils/rest-api.js [hunk 6]
```diff
@@ -111,6 +111,37 @@ export function getVoltageLevelSingleLineDiagram(
...
+    useName,
+    centerLabel,
+    diagonalLabel,
+    substationLayout
+) {
+    console.info(
+        `Getting url of substation diagram '${substationId}' of study '${studyName}' of user '${userId}'...`
+    );
+    return (
+        PREFIX_STUDY_QUERIES +
```
Comment (geofjamg) [inline src/utils/rest-api.js:127]: Coud you use the new getStudyUrl function?

Comment (FranckLecuyer) [reply src/utils/rest-api.js]: Done

**Pull request merged**